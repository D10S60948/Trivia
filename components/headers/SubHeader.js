import React, { Component } from 'react'
import { StyleSheet, View, Text, Image } from 'react-native'

export default class SubHeader extends Component {

    getImage() {
        if(this.props.text.length != 1) {
            return <Image source={require('../../images/logo3.png')} style={styles.image1} resizeMode='stretch' />
        } else {
            return <Image source={require('../../images/logo1.png')} style={styles.image2} resizeMode='stretch' />
        }
    }

    render() {
        return (
            <View style={styles.container}>    
                { this.getImage() }
                <View style={styles.container}>
                    {
                        this.props.text.length == 1 ?
                            <Text style={{...styles.text, fontSize: 60}}>{this.props.text[0]}</Text>
                        : 
                            <View style={styles.container}>    
                                <Text style={{...styles.text, fontSize: 50, marginBottom: -30}}>{this.props.text[0]}</Text>
                                <Text style={{...styles.text, fontSize: 60}}>{this.props.text[1]}</Text>
                            </View>
                    }
                </View>
            </View>      
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        flexDirection: 'column'
    },
    text: { 
      fontFamily: 'Modak',
      textShadowOffset: {width: 2,height:1},
      color:'#1F1F3F'
    },
    image1: {
        position: 'absolute',
        left: 5,
        top: 135,
        width: 80, 
        height: 73
    },
    image2: {
        position: 'absolute',
        right: 1,
        top: 58,
        width: 76, 
        height: 100,
        zIndex: 10
    }
});