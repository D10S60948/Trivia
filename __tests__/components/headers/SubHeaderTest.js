import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SubHeader from '../../../components/headers/SubHeader';

test('"SubHeader" component renders correctly', () => {
    const snapshot = renderer.create(<SubHeader text={['SETTINGS']}/>).toJSON();
    expect(snapshot).toMatchSnapshot();
});
