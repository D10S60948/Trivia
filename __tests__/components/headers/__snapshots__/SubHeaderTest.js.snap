// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"SubHeader" component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="stretch"
    source={
      Object {
        "testUri": "../../../images/logo1.png",
      }
    }
    style={
      Object {
        "height": 100,
        "position": "absolute",
        "right": 1,
        "top": 58,
        "width": 76,
        "zIndex": 10,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F3F",
          "fontFamily": "Modak",
          "fontSize": 60,
          "textShadowOffset": Object {
            "height": 1,
            "width": 2,
          },
        }
      }
    >
      SETTINGS
    </Text>
  </View>
</View>
`;
